File: accsaber-database/src/main/java/de/ixsen/accsaber/database/model/players/PlayerRankHistoryPK.java
package de.ixsen.accsaber.database.model.players;

import java.io.Serializable;
import java.time.LocalDate;
import java.util.Objects;

public class PlayerRankHistoryPK implements Serializable {

    protected Long player;
    protected LocalDate date;
    protected Long category;

    public PlayerRankHistoryPK() {
    }

    public PlayerRankHistoryPK(Long player, LocalDate date) {
        this.player = player;
        this.date = date;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || this.getClass() != o.getClass()) return false;
        PlayerRankHistoryPK that = (PlayerRankHistoryPK) o;
        return this.player.equals(that.player) && this.date.equals(that.date) && this.category.equals(that.category);
    }

    @Override
    public int hashCode() {
        return Objects.hash(this.player, this.date, this.category);
    }
}

File: accsaber-database/src/main/kotlin/de/ixsen/accsaber/database/model/players/PlayerRankHistoryPK.kt
package de.ixsen.accsaber.database.model.players

import java.io.Serializable
import java.time.LocalDate
import java.util.*

class PlayerRankHistoryPK : Serializable {
    protected var player: Long? = null
    protected var date: LocalDate? = null
    protected var category: Long? = null

    constructor() {}
    constructor(player: Long?, date: LocalDate?) {
        this.player = player
        this.date = date
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || this.javaClass != o.javaClass) return false
        val that = o as PlayerRankHistoryPK
        return player == that.player && date == that.date && this.category == that.category
    }

    override fun hashCode(): Int {
        return Objects.hash(player, date, this.category)
    }
}